### PR TITLE
Optimize snap install time 1

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
This is an RFC that improves snap installation time in two ways:

1. Removes unneeded calls to `systemctl daemon-reload`. See https://forum.snapcraft.io/t/are-systemctl-daemon-reload-calls-before-enabling-a-service-needed/6086 for more information.
2. It makes sure to call `apparmor_parser` only once when we want to (un)load a set of profiles, instead of looping on them.

### CODE REVIEW

File: interfaces/apparmor/apparmor.go [hunk 1]
```diff
@@ -37,46 +37,69 @@ import (
...
 
-func loadProfile(fname, cacheDir string) error {
+func loadProfiles(fnames []string, cacheDir string) error {
+	if len(fnames) == 0 {
+		return nil
+	}
+
 	// Use no-expr-simplify since expr-simplify is actually slower on armhf (LP: #1383858)
-	args := []string{"--replace", "--write-cache", "-O", "no-expr-simplify", fmt.Sprintf("--cache-loc=%s", cacheDir)}
+	args := []string{"--replace", "--write-cache", "-O", "no-expr-simplify",
```
Comment (zyga) [inline interfaces/apparmor/apparmor.go]: Is this a net win if we load multiple profiles in one go?

File: interfaces/apparmor/apparmor.go [hunk 2]
```diff
@@ -37,46 +37,69 @@ import (
...
 )
 
-// LoadProfile loads an apparmor profile from the given file.
+// LoadProfile loads apparmor profiles from the given files.
 //
 // If no such profile was previously loaded then it is simply added to the kernel.
 // If there was a profile with the same name before, that profile is replaced.
-func LoadProfile(fname string) error {
-	return loadProfile(fname, dirs.AppArmorCacheDir)
+func LoadProfiles(fnames []string) error {
```
Comment (zyga) [inline interfaces/apparmor/apparmor.go]: Suggestion, use `LoadManyProfiles` and perhaps use varadic arguments (less annoying in testing)

File: interfaces/apparmor/apparmor.go (hunk 1 above)
Comment (alfonsosanchezbeato) [reply interfaces/apparmor/apparmor.go]: It is, especially in slow ARM processors (a few seconds per set in an imx6ull).

**Comment by alfonsosanchezbeato** [comment]
Closing to propose 2 different MPs later.

**Comment by alfonsosanchezbeato** [comment]
First follow-up: https://github.com/snapcore/snapd/pull/5469

**Pull request closed without merging**